packages/open-next/src/overrides/cdnInvalidation/cloudfront.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  CloudFrontClient,
+  CreateInvalidationCommand,
+} from "@aws-sdk/client-cloudfront";
+import type { CDNInvalidationHandler } from "types/overrides";
+
+const cloudfront = new CloudFrontClient({});
+export default {
+  name: "cloudfront",
+  invalidatePaths: async (paths) => {
+    //TODO: test the constructed paths
+    const constructedPaths = paths.flatMap(({ path, isAppRouter }) =>
+      isAppRouter
+        ? [`${path}`, `${path}?_rsc=*`]
```

Comment:
I think only [`${path}*`] is enough here? I will do some testing tommorow btw.

Reply:
It's on purpose ` ${path}*` would invalidate every route that start with `path`. For example if you have a `products` and a `products/[slug]` route (using `${path}*`) it would  be impossible to invalidate only `products`

Reply:
I think it also invalidates the `products` if you use the `${path}*` one actually.

Reply:
That's not what i'm saying, it invalidates `products` but also every single `products/[slug]` route, which may not be what you want

Reply:
but yeah im guessing we dont want that. its probably wise to only invalidate the path.

in my projects im just used to doing the star, cause normally you would want `products` and `products/[slug]` invalidated at the same time.

Reply:
However, after ive done some testing this morning the paths need to have a leading `/` else an error will be thrown: `Failed to revalidate tag InvalidArgument: Your request contains one or more invalid invalidation paths.`

Reply:
The paths need to have a leading `/` else an error will be thrown: `Failed to revalidate tag InvalidArgument: Your request contains one or more invalid invalidation paths.`

Reply:
@sommeeeer It's only for `revalidateTag` right ? You didn't test `res.revalidate` ?

Reply:
> @sommeeeer It's only for `revalidateTag` right ? You didn't test `res.revalidate` ?
yeah, i will test `res.revalidate` now

Reply:
> It's only for `revalidateTag` right ? You didn't test `res.revalidate` ?

i tested `revalidatePath('/products');`

Reply:
This works:
```ts
const constructedPaths = paths.flatMap(({ path, isAppRouter }) =>
  isAppRouter
    ? [`/${path}`, `/${path}?_rsc=*`]
    : [
        `/${path}`,
        `/_next/data/${process.env.BUILD_ID}${path === "/" ? "/index" : path}.json*`,
      ],
);```